src/bin/pg_autoctl/cli_show.c

```diff
@@ -1186,35 +1196,114 @@ cli_show_file(int argc, char **argv)
 	{
 		case SHOW_FILE_ALL:
 		{
-			char *serialized_string = NULL;
-			JSON_Value *js = json_value_init_object();
-			JSON_Object *root = json_value_get_object(js);
+			if (outputJSON)
+			{
+				char *serialized_string = NULL;
+				JSON_Value *js = json_value_init_object();
+				JSON_Object *root = json_value_get_object(js);
 
-			json_object_set_string(root, "config", config.pathnames.config);
+				json_object_set_string(root, "config", config.pathnames.config);
 
-			if (role == PG_AUTOCTL_ROLE_KEEPER)
-			{
-				json_object_set_string(root, "state", config.pathnames.state);
-				json_object_set_string(root, "init", config.pathnames.init);
-			}
+				if (role == PG_AUTOCTL_ROLE_KEEPER)
+				{
+					json_object_set_string(root, "state", config.pathnames.state);
+					json_object_set_string(root, "init", config.pathnames.init);
+				}
 
-			json_object_set_string(root, "pid", config.pathnames.pid);
+				json_object_set_string(root, "pid", config.pathnames.pid);
 
-			serialized_string = json_serialize_to_string_pretty(js);
+				serialized_string = json_serialize_to_string_pretty(js);
 
-			fformat(stdout, "%s\n", serialized_string);
+				fformat(stdout, "%s\n", serialized_string);
 
-			json_free_serialized_string(serialized_string);
-			json_value_free(js);
+				json_free_serialized_string(serialized_string);
+				json_value_free(js);
+			}
+			else
+			{
+				fformat(stdout, "%6s | %s\n", "File", "Path");
+				fformat(stdout, "%6s-+-%15s\n", "------", "---------------");
 
+				fformat(stdout, "%6s | %s\n", "Config", config.pathnames.config);
+
+				if (role == PG_AUTOCTL_ROLE_KEEPER)
+				{
+					fformat(stdout, "%6s | %s\n", "State", config.pathnames.state);
+					fformat(stdout, "%6s | %s\n", "Init", config.pathnames.init);
+				}
+				fformat(stdout, "%6s | %s\n", "Pid", config.pathnames.pid);
+				fformat(stdout, "\n");
```

Comment:
nitpick: I think the table could look slightly nicer. Instead of this:
```
  File | Path
-------+----------------
Config | /home/jelte/.config/pg_autoctl/home/jelte/work/pg_auto_failover/monitor/pg_autoctl.cfg
   Pid | /tmp/pg_autoctl/home/jelte/work/pg_auto_failover/monitor/pg_autoctl.pid
```
It would look like this (note the extra space in front of "Config"):
```
   File | Path
--------+----------------
 Config | /home/jelte/.config/pg_autoctl/home/jelte/work/pg_auto_failover/monitor/pg_autoctl.cfg
    Pid | /tmp/pg_autoctl/home/jelte/work/pg_auto_failover/monitor/pg_autoctl.pid
```
Even nicer would be if the header line of `Path` would match up with the contents, i.e. getting longest length. But I do not think this is worth the effort tbh.

Reply:
Fair enough, done! Well not the Path parts which I agree with you doesn't seem worth the effort, and also could be ugly if the header line spans larger than the terminal COLUMNS etc.